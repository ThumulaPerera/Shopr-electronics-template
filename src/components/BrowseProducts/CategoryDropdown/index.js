import React from 'react';
import {
  Dropdown,
} from 'semantic-ui-react';
import { isEmpty } from 'react-redux-firebase';
import { compose } from 'redux';
import { useRouteMatch } from 'react-router-dom';
import PropTypes from 'prop-types';

import applyUrlCorrection from '../../../helpers/applyUrlCorrection';

const categoryDropdown = (props) => {
  const {
    categories, handleCategoryClick, selectedCategory,
  } = props;

  if (isEmpty(categories)) {
    return null;
  }

  const options = [];
  options.push({ key: 'All', text: 'All', value: 'All' });
  Object.keys(categories).map((key) => {
    const { name } = categories[key];
    options.push({ key: name, text: name, value: name });
    return null;
  });

  return (
    <Dropdown
      onChange={handleCategoryClick}
      options={options}
      search
      selection
      value={selectedCategory}
      scrolling
      fluid
    />
  );
};

function withHooks(Component) {
  return function WrappedComponent(props) {
    const { url } = useRouteMatch();
    return <Component {...props} url={applyUrlCorrection(url)} />;
  };
}

export default compose(
  withHooks,
)(categoryDropdown);

categoryDropdown.propTypes = {
  categories: PropTypes.array.isRequired,
  selectedCategory: PropTypes.string.isRequired,
  handleCategoryClick: PropTypes.func.isRequired,
};
