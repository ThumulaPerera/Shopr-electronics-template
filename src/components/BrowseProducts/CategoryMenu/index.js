import React from 'react';
import {
  Menu, Segment, Sticky, Header, Icon,
} from 'semantic-ui-react';

import { isEmpty } from 'react-redux-firebase';
import { compose } from 'redux';
import { useRouteMatch } from 'react-router-dom';
import PropTypes from 'prop-types';

import applyUrlCorrection from '../../../helpers/applyUrlCorrection';

const CategoryMenu = (props) => {
  const {
    categories, contextRef, storeCustomization, handleCategoryClick, selectedCategory,
  } = props;

  if (isEmpty(categories)) {
    return <div>No categories to display...</div>;
  }

  let color;
  if (storeCustomization) {
    color = storeCustomization.color;
  }

  return (
    <Sticky
      context={contextRef}
      offset={140}
      pushing
    >
      <Segment padded basic>
        <Header>
          <Icon name="list alternate outline" />
          <Header.Content>Categories</Header.Content>
        </Header>
        <div style={{ overflowY: 'auto', overflowX: 'visible', maxHeight: '40vh' }}>
          <Menu secondary pointing vertical color={color}>
            <Menu.Item
              as="a"
              name="All"
              active={selectedCategory === 'All'}
              onClick={handleCategoryClick}
            />
            {Object.keys(categories).map((key) => {
              const { name } = categories[key];
              return (
                <Menu.Item
                  as="a"
                  name={name}
                  active={selectedCategory === name}
                  onClick={handleCategoryClick}
                  key={key}
                />
              );
            })}
          </Menu>
        </div>
      </Segment>
    </Sticky>

  );
};

function withHooks(Component) {
  return function WrappedComponent(props) {
    const { url } = useRouteMatch();
    return <Component {...props} url={applyUrlCorrection(url)} />;
  };
}

export default compose(
  withHooks,
)(CategoryMenu);

CategoryMenu.propTypes = {
  categories: PropTypes.array.isRequired,
  contextRef: PropTypes.object.isRequired,
  selectedCategory: PropTypes.string.isRequired,
  handleCategoryClick: PropTypes.func.isRequired,
  storeCustomization: PropTypes.object.isRequired,
};
