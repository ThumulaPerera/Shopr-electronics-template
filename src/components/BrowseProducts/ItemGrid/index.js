import React from 'react';
import {
  Card, Segment, Header, Icon,
} from 'semantic-ui-react';
import { connect } from 'react-redux';
import { isLoaded, isEmpty } from 'react-redux-firebase';
import { get } from 'lodash';
import { compose } from 'redux';
import { useParams } from 'react-router-dom';
import PropTypes from 'prop-types';

import ItemCard from '../../ItemCard';
import calculateRating from '../../../helpers/calculateRating';
import getItemsAndIconByCategory from '../../../helpers/getItemsAndIconByCategory';

const ItemGrid = (props) => {
  let { items } = props;
  const {
    categories, selectedCategory, url, currency, searchString,
  } = props;
  const defaultImgUrl = 'https://www.cowgirlcontractcleaning.com/wp-content/uploads/sites/360/2018/05/placeholder-img-1.jpg';
  let icon;

  if (!(isLoaded(items) && isLoaded(categories))) {
    return <div>Loading...</div>;
  }

  if (isLoaded(items) && isLoaded(categories)) {
    const filtered = getItemsAndIconByCategory(items, categories, selectedCategory);
    items = filtered.itemsOfSelectedCategory;
    icon = filtered.categoryIcon;
  }

  if (!items || isEmpty(items)) {
    return (
      <Segment basic>
        <Header as="h2">
          <Header.Content>
            {icon ? <Icon name={icon} /> : null}
            {selectedCategory}
          </Header.Content>
        </Header>
        <div>No items to display....</div>
      </Segment>
    );
  }

  return (

    <Segment basic padded="very">
      <Header as="h2">
        <Header.Content>
          {icon ? <Icon name={icon} /> : null}
          {selectedCategory}
        </Header.Content>
      </Header>
      <Card.Group itemsPerRow="3" stackable>
        {Object.keys(items).map((itemKey) => {
          const {
            name, photos, description, basePrice, rating, discount,
          } = items[itemKey];

          // eslint-disable-next-line no-nested-ternary
          const imageURL = photos && photos[0]
            ? (photos[0].thumbnail ? photos[0].thumbnail : photos[0].url)
            : defaultImgUrl;
          const amount = basePrice;
          const itemRating = rating ? calculateRating(rating) : null;

          const minRequirementsToDisplay = name && basePrice && name.search(new RegExp(searchString, 'i')) !== -1;

          if (!minRequirementsToDisplay) {
            return null;
          }

          return (
            <ItemCard
              key={itemKey}
              id={itemKey}
              name={name}
              imageURL={imageURL}
              description={description}
              currency={currency}
              price={amount}
              rating={itemRating}
              discount={discount}
              url={url}
            />
          );
        })}
      </Card.Group>
    </Segment>
  );
};

const mapStateToProps = (state) => ({
  categories: get(state.firestore.data, 'sellerStore.categories'),
  currency: get(state.firestore.data, 'sellerStore.currency'),
  items: get(state.firestore.data, 'sellerItems'),
});

function withHooks(Component) {
  return function WrappedComponent(props) {
    const { storeID, category } = useParams();
    return (
      <Component
        {...props}
        storeID={storeID}
        // eslint-disable-next-line
        selectedCategory={category || props.selectedCategory}
      />
    );
  };
}

export default compose(
  withHooks,
  connect(mapStateToProps),
)(ItemGrid);

ItemGrid.propTypes = {
  categories: PropTypes.array.isRequired,
  items: PropTypes.object,
  currency: PropTypes.string.isRequired,
  searchString: PropTypes.string.isRequired,
  selectedCategory: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

ItemGrid.defaultProps = {
  items: undefined,
};
