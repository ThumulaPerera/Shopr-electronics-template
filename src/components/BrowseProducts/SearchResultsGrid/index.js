import React from 'react';
import {
  Card, Segment,
} from 'semantic-ui-react';
import { isEmpty } from 'react-redux-firebase';
import PropTypes from 'prop-types';

import ItemCard from '../../ItemCard';
import calculateRating from '../../../helpers/calculateRating';

const ItemGrid = (props) => {
  const {
    items, url, currency, searchString,
  } = props;
  const defaultImgUrl = 'https://www.cowgirlcontractcleaning.com/wp-content/uploads/sites/360/2018/05/placeholder-img-1.jpg';


  if (!items || isEmpty(items)) {
    return null;
  }

  const filteredItems = items.filter((item) => (
    item.name.search(new RegExp(searchString, 'i')) !== -1
  ));

  if (!filteredItems || isEmpty(filteredItems)) {
    return null;
  }

  return (
    <Segment basic padded="very">
      <Card.Group itemsPerRow="3" stackable>
        {Object.keys(filteredItems).map((itemKey) => {
          const {
            name, photos, description, basePrice, rating, discount, id,
          } = filteredItems[itemKey];

          // eslint-disable-next-line no-nested-ternary
          const imageURL = photos && photos[0]
            ? (photos[0].thumbnail ? photos[0].thumbnail : photos[0].url)
            : defaultImgUrl;
          const amount = basePrice;
          const itemRating = rating ? calculateRating(rating) : null;

          const minRequirementsToDisplay = name && basePrice;

          if (!minRequirementsToDisplay) {
            return null;
          }

          return (
            <ItemCard
              key={id}
              id={id}
              name={name}
              imageURL={imageURL}
              description={description}
              currency={currency}
              price={amount}
              rating={itemRating}
              discount={discount}
              url={url}
            />
          );
        })}
      </Card.Group>
    </Segment>
  );
};

export default ItemGrid;

ItemGrid.propTypes = {
  items: PropTypes.array.isRequired,
  currency: PropTypes.string.isRequired,
  searchString: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
