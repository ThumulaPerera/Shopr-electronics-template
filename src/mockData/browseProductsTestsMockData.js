/* eslint-disable import/prefer-default-export */

export const minItems = [
  {
    id: '5ed1112909f2ff2d85b2c21b',
    basePrice: 3835.16,
    category: 3,
    deleted: true,
    description: 'Sint elit dolore pariatur in occaecat pariatur amet est officia elit ea enim. Enim amet enim elit ut nulla sit in minim eu ea. Consequat duis anim laboris laborum officia exercitation amet reprehenderit deserunt consequat fugiat ea aute ex. Ex deserunt labore in sunt ex proident sit ad. Ad elit excepteur irure eiusmod nisi Lorem eu qui ipsum ea. Ipsum excepteur non et velit occaecat duis ut.\r\n',
    name: 'fugiat',
    visible: true,
  },
  {
    id: '5ed111295a728d32a9ca49b7',
    basePrice: 1684.05,
    category: 2,
    deleted: false,
    description: 'Velit cillum veniam proident in deserunt voluptate cillum. Exercitation velit cillum id commodo aliquip dolor reprehenderit dolore consectetur. Non dolor do cillum nisi. Exercitation sint consectetur irure culpa laboris Lorem reprehenderit sit occaecat occaecat do aute esse. Irure minim nostrud sint adipisicing ipsum eu aliquip.\r\n',
    name: 'dolor',
    visible: true,
  },
  {
    id: '5ed1112908cf459fc5101684',
    basePrice: 1921.53,
    category: 0,
    deleted: true,
    description: 'Quis commodo occaecat ut in ad quis. Anim eiusmod dolore Lorem sunt eu sint adipisicing officia. Officia laboris ad dolor esse aliqua.\r\n',
    name: 'laborum',
    visible: false,
  },
  {
    id: '5ed111296d4090b28f206d38',
    basePrice: 3172.05,
    category: 2,
    deleted: false,
    description: 'Id fugiat consectetur proident nisi sunt quis sint duis minim adipisicing duis. Do enim amet do sit adipisicing elit qui consectetur. Elit laborum ullamco amet eu magna pariatur labore ad ea adipisicing ullamco. Cillum non aliqua aliqua ullamco voluptate voluptate ipsum nulla officia sunt. Excepteur velit aliquip ipsum pariatur laborum qui aliqua sunt. Ullamco pariatur consectetur amet dolor sit.\r\n',
    name: 'ipsum',
    visible: true,
  },
  {
    id: '5ed11129a3c7710259fa19e1',
    basePrice: 4388.14,
    category: 2,
    deleted: false,
    description: 'Est cupidatat est adipisicing est enim magna sit duis ad culpa Lorem labore. Nulla commodo laboris ut excepteur labore do ipsum sit proident officia. Elit esse non in minim in. Laboris laboris dolore tempor mollit consequat culpa id amet. Aliquip commodo cupidatat duis ad ut ex cupidatat in et sunt officia.\r\n',
    name: 'officia',
    visible: false,
  },
  {
    id: '5ed11129fb522bfdb34ebe21',
    basePrice: 3317.43,
    category: 3,
    deleted: false,
    description: 'Lorem veniam tempor est fugiat proident sit irure nisi veniam nostrud. Dolor id fugiat laboris elit cupidatat aute ad labore eu fugiat. Enim voluptate irure id minim amet anim occaecat officia sint pariatur laboris velit non consectetur. Pariatur est mollit sunt id laboris mollit adipisicing voluptate culpa eiusmod magna nisi. Pariatur dolore labore magna enim aute qui velit officia adipisicing sit ad. Laboris dolor nisi adipisicing veniam est veniam duis labore id ipsum dolor eiusmod. Est amet ex ex adipisicing in amet sit anim magna aliqua.\r\n',
    name: 'mollit',
    visible: false,
  },
  {
    id: '5ed11129f38a4da11e536dec',
    basePrice: 3623.22,
    category: 3,
    deleted: false,
    description: 'Cillum nulla nulla commodo sint laborum aliquip irure proident mollit dolore nostrud laboris irure. In mollit tempor fugiat aute nisi pariatur labore nostrud eiusmod duis. Laborum aute laborum do sit duis est excepteur sit Lorem eu dolor pariatur. In ea deserunt amet do adipisicing. Exercitation officia irure labore commodo voluptate adipisicing sit nisi consectetur occaecat non magna nisi. Fugiat qui ipsum nulla nisi eu et dolor tempor proident cupidatat proident ipsum cillum culpa.\r\n',
    name: 'mollit',
    visible: false,
  },
  {
    id: '5ed11129bf7508bf980aaebc',
    basePrice: 3488.74,
    category: 2,
    deleted: true,
    description: 'Sit exercitation ullamco anim mollit dolor nulla ut nisi occaecat in. Occaecat ut qui culpa qui laboris sunt. Consequat proident et eiusmod magna. Ullamco occaecat voluptate qui id nulla minim non velit commodo. Nostrud enim sunt sint magna ullamco officia reprehenderit non cupidatat minim elit. Aliqua adipisicing ex nostrud aute eu do ullamco. Excepteur pariatur nisi qui occaecat nisi veniam in.\r\n',
    name: 'eiusmod',
    visible: true,
  },
  {
    id: '5ed11129636830919c3d2175',
    basePrice: 2480.81,
    category: 0,
    deleted: false,
    description: 'Irure veniam magna labore nisi aliquip reprehenderit aliquip. Reprehenderit ex eu nisi esse id commodo nisi exercitation sunt dolor ad officia consectetur. Dolor laboris occaecat Lorem et id cillum dolor occaecat nisi.\r\n',
    name: 'nulla',
    visible: false,
  },
  {
    id: '5ed111295cc737557bf875ce',
    basePrice: 2468.8,
    category: 3,
    deleted: false,
    description: 'Irure eiusmod officia cupidatat veniam. Exercitation anim laborum cupidatat tempor irure do. Ut ad sunt id nostrud eiusmod mollit mollit. Officia sint eu id exercitation nisi consequat id dolore.\r\n',
    name: 'excepteur',
    visible: true,
  },
];

export const sellerStore = {
  categories: [
    {
      name: 'category 0',
    },
    {
      name: 'category 1',
    },
    {
      name: 'category 2',
    },
    {
      name: 'category 3',
    },
    {
      name: 'category 4',
    },
  ],
  currency: 'USD',
  email: 'hendersonreed@orbiflex.com',
  enableChatbot: true,
  enableInventoryManagement: false,
  enableRating: true,
  verified: false,
  storeName: 'ut qui',
  storeCustomization: {},
};
