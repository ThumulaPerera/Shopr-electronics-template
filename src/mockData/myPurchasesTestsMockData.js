/* eslint-disable import/prefer-default-export */
export const items = {
  '5ed111295a728d32a9ca49b7': {
    id: '5ed111295a728d32a9ca49b7',
    basePrice: 100,
    category: 2,
    deleted: false,
    description: 'Velit cillum veniam proident in deserunt voluptate cillum. Exercitation velit cillum id commodo aliquip dolor reprehenderit dolore consectetur. Non dolor do cillum nisi. Exercitation sint consectetur irure culpa laboris Lorem reprehenderit sit occaecat occaecat do aute esse. Irure minim nostrud sint adipisicing ipsum eu aliquip.\r\n',
    name: 'dolor',
    visible: true,
    variants: [
      {
        title: 'variant title 1',
        attributes: [
          {
            attribute: 'variant attribute 11',
          },
          {
            attribute: 'variant attribute 12',
          },
        ],
      },
      {
        title: 'variant title 2',
        attributes: [
          {
            attribute: 'variant attribute 21',
          },
          {
            attribute: 'variant attribute 22',
          },
        ],
      },
    ],
    subItems: [
      {
        price: 100,
        stock: 10,
        variants: [
          'variant attribute 11',
          'variant attribute 21',
        ],
      },
      {
        price: 100,
        stock: 10,
        variants: [
          'variant attribute 12',
          'variant attribute 21',
        ],
      },
      {
        price: 100,
        stock: 10,
        variants: [
          'variant attribute 11',
          'variant attribute 22',
        ],
      },
      {
        price: 100,
        stock: 10,
        variants: [
          'variant attribute 12',
          'variant attribute 22',
        ],
      },
    ],
  },
  '5ed111296d4090b28f206d38': {
    id: '5ed111296d4090b28f206d38',
    basePrice: 3172.05,
    category: 2,
    deleted: false,
    description: 'Id fugiat consectetur proident nisi sunt quis sint duis minim adipisicing duis. Do enim amet do sit adipisicing elit qui consectetur. Elit laborum ullamco amet eu magna pariatur labore ad ea adipisicing ullamco. Cillum non aliqua aliqua ullamco voluptate voluptate ipsum nulla officia sunt. Excepteur velit aliquip ipsum pariatur laborum qui aliqua sunt. Ullamco pariatur consectetur amet dolor sit.\r\n',
    name: 'ipsum',
    visible: true,
  },
  '5ed111295cc737557bf875ce': {
    id: '5ed111295cc737557bf875ce',
    basePrice: 2468.8,
    category: 3,
    deleted: false,
    description: 'Irure eiusmod officia cupidatat veniam. Exercitation anim laborum cupidatat tempor irure do. Ut ad sunt id nostrud eiusmod mollit mollit. Officia sint eu id exercitation nisi consequat id dolore.\r\n',
    name: 'excepteur',
    visible: true,
  },
};

export const sellerStore = {
  categories: [
    {
      name: 'category 0',
    },
    {
      name: 'category 1',
    },
    {
      name: 'category 2',
    },
    {
      name: 'category 3',
    },
    {
      name: 'category 4',
    },
  ],
  currency: 'USD',
  email: 'hendersonreed@orbiflex.com',
  enableChatbot: true,
  enableInventoryManagement: true,
  enableRating: true,
  verified: true,
  storeName: 'ut qui',
  storeCustomization: {},
  merchantId: '123456',
};

export const orders = [
  {
    id: 'order123',
    orderStates: [
      {
        date: {
          seconds: 1590234479,
          nanoseconds: 947000000,
        },
        stateId: 0,
      },
    ],
    buyer: 'buyer123',
    paymentMethod: 'paypal',
    totalPrice: 3200,
    shippingAddress: {},
    orderItems: [

      {
        item: '5ed111295a728d32a9ca49b7',
        noOfItems: 1,
        subItem: 0,
        unitPrice: 100,
      },

    ],
    date: {
      seconds: 1590234479,
      nanoseconds: 947000000,
    },
  },
];
