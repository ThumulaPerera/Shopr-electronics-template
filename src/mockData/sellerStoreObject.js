/* eslint-disable import/prefer-default-export */
export const sellerStore = {
  categories: [
    {
      name: 'category 0',
    },
    {
      name: 'category 1',
    },
    {
      name: 'category 2',
    },
    {
      name: 'category 3',
    },
    {
      name: 'category 4',
    },
  ],
  currency: 'USD',
  email: 'hendersonreed@orbiflex.com',
  enableChatbot: true,
  enableInventoryManagement: false,
  enableRating: true,
  verified: false,
  storeName: 'ut qui',
  storeCustomization: {},
};
