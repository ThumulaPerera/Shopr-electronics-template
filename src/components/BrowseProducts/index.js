import React from 'react';
import { Grid } from 'semantic-ui-react';

import { useRouteMatch, Switch, Route } from 'react-router-dom';
import { get } from 'lodash';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { isLoaded, isEmpty } from 'react-redux-firebase'

import ItemGrid from '../ItemGrid';
import CategoryMenu from '../CategoryMenu';

const BrowseProducts = ({ contextRef, categories }) => {
    const { path } = useRouteMatch()

    if (!isLoaded(categories)) {
        return <div>Loading...</div>
    }

    if (isEmpty(categories)) {
        return <div>No categories to display...</div>
    }

    return(
    <Grid columns={2} divided>
        <Grid.Row>
            <Grid.Column width={3}>
                <CategoryMenu contextRef={contextRef} />
            </Grid.Column>
            <Grid.Column width={13}>
                <Switch>
                    <Route exact path={`${path}/`}>
                            {
                                categories.map(({ name }) => (
                                    <div>
                                        <ItemGrid selectedCategory={name} />
                                    </div>
                                ))
                            }
                    </Route>
                    <Route path={`${path}/:category`}>
                        <ItemGrid />
                    </Route>
                </Switch>
            </Grid.Column>
        </Grid.Row>
    </Grid>
)}

const mapStateToProps = (state) => ({
    categories: get(state.firestore.data, `sellerStore.categories`),
});


export default compose(
    connect(mapStateToProps),
)(BrowseProducts);
