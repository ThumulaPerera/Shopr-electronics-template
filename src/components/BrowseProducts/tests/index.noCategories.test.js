/* eslint-disable no-undef */
import React, { createRef } from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { cleanup } from '@testing-library/react';
import { mount } from 'enzyme';
import { Route, MemoryRouter } from 'react-router-dom';

import BrowseProducts, { BrowseProducts as UnconnectedBrowseProducts } from '../index';

import { minItems, sellerStore } from '../../../mockData/browseProductsTestsMockData';


jest.unmock('react-redux-firebase');
const rrf = require.requireActual('react-redux-firebase');

const mockStore = configureStore([]);

let store;
let component;

beforeEach(() => {
  store = mockStore({
    firestore: {
      ordered: {
        sellerItems: minItems,
      },
      data: {
        sellerStore,
      },
    },
  });

  // mock fn for window.scrollTo
  global.scrollTo = jest.fn();
  // mock fn for isLoaded
  rrf.isLoaded = jest.fn(() => true);
  // mock fn for isEmpty
  rrf.isEmpty = jest.fn(() => true);

  // const ref = createRef();
  const wrapperComponent = mount(
    <Provider store={store}>
      <MemoryRouter initialEntries={[{ pathname: '/home', key: 'testKey' }]}>
        <Route path="/home">
          <BrowseProducts contextRef={createRef()} />
        </Route>
      </MemoryRouter>
    </Provider>,
  );
  component = wrapperComponent.find(UnconnectedBrowseProducts);
});

afterEach(cleanup);


it('renders no categories to display message if categories are empty', () => {
  expect(component).toMatchSnapshot();

  // renders no categories message
  const message = (
    <div>
      No categories to display...
    </div>
  );
  expect(component).toContainReact(message);
});
