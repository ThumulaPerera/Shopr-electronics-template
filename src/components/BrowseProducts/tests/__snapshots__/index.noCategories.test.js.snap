// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders no categories to display message if categories are empty 1`] = `
<BrowseProducts
  categories={
    Array [
      Object {
        "name": "category 0",
      },
      Object {
        "name": "category 1",
      },
      Object {
        "name": "category 2",
      },
      Object {
        "name": "category 3",
      },
      Object {
        "name": "category 4",
      },
    ]
  }
  contextRef={
    Object {
      "current": null,
    }
  }
  currency="USD"
  dispatch={[Function]}
  items={
    Array [
      Object {
        "basePrice": 3835.16,
        "category": 3,
        "deleted": true,
        "description": "Sint elit dolore pariatur in occaecat pariatur amet est officia elit ea enim. Enim amet enim elit ut nulla sit in minim eu ea. Consequat duis anim laboris laborum officia exercitation amet reprehenderit deserunt consequat fugiat ea aute ex. Ex deserunt labore in sunt ex proident sit ad. Ad elit excepteur irure eiusmod nisi Lorem eu qui ipsum ea. Ipsum excepteur non et velit occaecat duis ut.
",
        "id": "5ed1112909f2ff2d85b2c21b",
        "name": "fugiat",
        "visible": true,
      },
      Object {
        "basePrice": 1684.05,
        "category": 2,
        "deleted": false,
        "description": "Velit cillum veniam proident in deserunt voluptate cillum. Exercitation velit cillum id commodo aliquip dolor reprehenderit dolore consectetur. Non dolor do cillum nisi. Exercitation sint consectetur irure culpa laboris Lorem reprehenderit sit occaecat occaecat do aute esse. Irure minim nostrud sint adipisicing ipsum eu aliquip.
",
        "id": "5ed111295a728d32a9ca49b7",
        "name": "dolor",
        "visible": true,
      },
      Object {
        "basePrice": 1921.53,
        "category": 0,
        "deleted": true,
        "description": "Quis commodo occaecat ut in ad quis. Anim eiusmod dolore Lorem sunt eu sint adipisicing officia. Officia laboris ad dolor esse aliqua.
",
        "id": "5ed1112908cf459fc5101684",
        "name": "laborum",
        "visible": false,
      },
      Object {
        "basePrice": 3172.05,
        "category": 2,
        "deleted": false,
        "description": "Id fugiat consectetur proident nisi sunt quis sint duis minim adipisicing duis. Do enim amet do sit adipisicing elit qui consectetur. Elit laborum ullamco amet eu magna pariatur labore ad ea adipisicing ullamco. Cillum non aliqua aliqua ullamco voluptate voluptate ipsum nulla officia sunt. Excepteur velit aliquip ipsum pariatur laborum qui aliqua sunt. Ullamco pariatur consectetur amet dolor sit.
",
        "id": "5ed111296d4090b28f206d38",
        "name": "ipsum",
        "visible": true,
      },
      Object {
        "basePrice": 4388.14,
        "category": 2,
        "deleted": false,
        "description": "Est cupidatat est adipisicing est enim magna sit duis ad culpa Lorem labore. Nulla commodo laboris ut excepteur labore do ipsum sit proident officia. Elit esse non in minim in. Laboris laboris dolore tempor mollit consequat culpa id amet. Aliquip commodo cupidatat duis ad ut ex cupidatat in et sunt officia.
",
        "id": "5ed11129a3c7710259fa19e1",
        "name": "officia",
        "visible": false,
      },
      Object {
        "basePrice": 3317.43,
        "category": 3,
        "deleted": false,
        "description": "Lorem veniam tempor est fugiat proident sit irure nisi veniam nostrud. Dolor id fugiat laboris elit cupidatat aute ad labore eu fugiat. Enim voluptate irure id minim amet anim occaecat officia sint pariatur laboris velit non consectetur. Pariatur est mollit sunt id laboris mollit adipisicing voluptate culpa eiusmod magna nisi. Pariatur dolore labore magna enim aute qui velit officia adipisicing sit ad. Laboris dolor nisi adipisicing veniam est veniam duis labore id ipsum dolor eiusmod. Est amet ex ex adipisicing in amet sit anim magna aliqua.
",
        "id": "5ed11129fb522bfdb34ebe21",
        "name": "mollit",
        "visible": false,
      },
      Object {
        "basePrice": 3623.22,
        "category": 3,
        "deleted": false,
        "description": "Cillum nulla nulla commodo sint laborum aliquip irure proident mollit dolore nostrud laboris irure. In mollit tempor fugiat aute nisi pariatur labore nostrud eiusmod duis. Laborum aute laborum do sit duis est excepteur sit Lorem eu dolor pariatur. In ea deserunt amet do adipisicing. Exercitation officia irure labore commodo voluptate adipisicing sit nisi consectetur occaecat non magna nisi. Fugiat qui ipsum nulla nisi eu et dolor tempor proident cupidatat proident ipsum cillum culpa.
",
        "id": "5ed11129f38a4da11e536dec",
        "name": "mollit",
        "visible": false,
      },
      Object {
        "basePrice": 3488.74,
        "category": 2,
        "deleted": true,
        "description": "Sit exercitation ullamco anim mollit dolor nulla ut nisi occaecat in. Occaecat ut qui culpa qui laboris sunt. Consequat proident et eiusmod magna. Ullamco occaecat voluptate qui id nulla minim non velit commodo. Nostrud enim sunt sint magna ullamco officia reprehenderit non cupidatat minim elit. Aliqua adipisicing ex nostrud aute eu do ullamco. Excepteur pariatur nisi qui occaecat nisi veniam in.
",
        "id": "5ed11129bf7508bf980aaebc",
        "name": "eiusmod",
        "visible": true,
      },
      Object {
        "basePrice": 2480.81,
        "category": 0,
        "deleted": false,
        "description": "Irure veniam magna labore nisi aliquip reprehenderit aliquip. Reprehenderit ex eu nisi esse id commodo nisi exercitation sunt dolor ad officia consectetur. Dolor laboris occaecat Lorem et id cillum dolor occaecat nisi.
",
        "id": "5ed11129636830919c3d2175",
        "name": "nulla",
        "visible": false,
      },
      Object {
        "basePrice": 2468.8,
        "category": 3,
        "deleted": false,
        "description": "Irure eiusmod officia cupidatat veniam. Exercitation anim laborum cupidatat tempor irure do. Ut ad sunt id nostrud eiusmod mollit mollit. Officia sint eu id exercitation nisi consequat id dolore.
",
        "id": "5ed111295cc737557bf875ce",
        "name": "excepteur",
        "visible": true,
      },
    ]
  }
  match={
    Object {
      "isExact": true,
      "params": Object {},
      "path": "/home",
      "url": "/home",
    }
  }
  ratingEnabled={true}
  storeCustomization={Object {}}
  storeName="ut qui"
>
  <div>
    <HelmetWrapper
      defer={true}
      encodeSpecialCharacters={true}
    >
      <SideEffect(NullComponent)
        defer={true}
        encodeSpecialCharacters={true}
        title={
          Array [
            "Products -",
            " ",
            "ut qui",
          ]
        }
        titleAttributes={Object {}}
      >
        <NullComponent
          defer={true}
          encodeSpecialCharacters={true}
          title={
            Array [
              "Products -",
              " ",
              "ut qui",
            ]
          }
          titleAttributes={Object {}}
        />
      </SideEffect(NullComponent)>
    </HelmetWrapper>
    <div>
      No categories to display...
    </div>
  </div>
</BrowseProducts>
`;
